Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/details/VaccinationDetailsFragment.kt

```diff
@@ -61,9 +61,11 @@ class VaccinationDetailsFragment : Fragment(R.layout.fragment_vaccination_detail
 
             viewModel.errors.observe(viewLifecycleOwner) { it.toErrorDialogBuilder(requireContext()).show() }
             viewModel.qrCode.observe(viewLifecycleOwner) {
-                qrCodeCard.progressBar.hide()
                 qrCodeCard.image.setImageBitmap(it)
-                it?.let { qrCodeCard.image.setOnClickListener { viewModel.openFullScreen() } }
+                it?.let {
+                    qrCodeCard.image.setOnClickListener { viewModel.openFullScreen() }
+                    qrCodeCard.progressBar.hide()
```

Comment:
Is that so? I thought, if there is no image, we just keep showing the indicator, the image might be emitted later.

Reply:
An error could be thrown and in this case, there is no image , and  a dialog will be shown , user can dismiss the dialog , no reason to keep the progress running.

Reply:
Null is emitted initially, then the image.

Reply:
Ah, you are right, but this is actually a bug. that is why the progress is hiding a little bit earlier before the image is displayed. Still I don't think we should show the progress when :
1- an exception is thrown -> no image 
2- No vaccination is matching -> no image 

I will replace the state flow with Livedata. the initial value was mainly because state flow requires in initial value.

Reply:
Thanks